### DISCUSSION TITLE
Issue with watchContractEvent (The request timed out)

### DISCUSSION THREAD

**Discussion by therawk** [wevm/viem#1441] [Question]
So I'm using the wagmi hooks for react but the error message shows that this is coming from viem:
```
292-a9b74eb148204e25.js:29 Uncaught (in promise) TimeoutError: The request took too long to respond.

URL: wss://ws.pulsechain.com/
Request body: {"method":"eth_uninstallFilter","params":["0x208b0100000000001c5f08aa63a10a3c"]}

Details: The request timed out.
Version: viem@1.18.4
    at Object.webSocketAsync (292-a9b74eb148204e25.js:29:21319)
    at request (292-a9b74eb148204e25.js:29:22602)
    at async withRetry.delay.count (292-a9b74eb148204e25.js:28:141938)
    at async attemptRetry (292-a9b74eb148204e25.js:28:141444)
```

Pretty hard to understand what's really going on here - thing is the websocket connection get closed over and over again:

<img width="731" alt="Screenshot 2023-11-03 at 14 48 10" src="https://github.com/wagmi-dev/viem/assets/24702753/efc28c33-6e87-4990-8391-6710679e9ded">

I saw that sometimes there are 3 open websocket connections at the same time. Has anybody ever faced this and found a proper solution?
Basically I just want to listen to 2 events on two contracts and sometimes refresh my page if a transaction has been found which matches my conditions.

Any ideas would be much appreciated
